### ISSUE TITLE
Implement Scroll Group Design

### ISSUE THREAD

**Issue by katherinejensen00** [paranext/paranext-core#788]
**User Story**
As a user, I want to have different tabs scroll together at different references so that I can view different texts at different places in Scripture for things like Parallel Passages or cross references.

**Description**
Implement UX Scroll Group Design.

**Comment by Sebastian-ubs** [comment]
related UX epic is https://paratext.myjetbrains.com/youtrack/issue/BUX-38/Epic-Scroll-groups-and-scripture-reference-control

**Comment by ianhewubs** [comment]
Related UX case for scroll groups is [BUX-17](https://paratext.myjetbrains.com/youtrack/issue/BUX-17). This contains links to mockups (currently within the BCV design mockups. UX hopes to split these out into a scroll group specific mockup).

**Comment by merchako** [comment]
@paranext/paranext-dev-write-access , notice two affordances for changing scroll group: (1) a full one in the Tab Options Menu and (2) a smaller one in the Tab Nav. I would call both "exclusive toggle button groups." For a starting point/comparison, see [Shadcn/ui's (independent) Toggle Group](https://ui.shadcn.com/docs/components/toggle-group#large).



![Image](https://github.com/paranext/paranext-core/assets/441789/7b209287-5b98-4bd9-a193-08f4d193f3e5)



* Similar to Shadcn Toggle Group but
* behaves exclusively
* choices are visually connected

![Image](https://github.com/paranext/paranext-core/assets/441789/ad3249ea-a6b3-4025-a2a6-1c154dd1cbc6)

The smaller one has some funky behavior. ⚠️ This is risky but we’re trying it out. UX would like you're opinion on whether it's worth developing it as a separate, unproven component or variation.

 * It hides away less useful choices by showing only populated scroll groups and the alphabetically leftmost empty scroll group.
 * Instead of showing a Null option, the user can click on the selected scroll group to deselect it. Thus, selection can be 1 option (radio) or none.

How hard would this variation be to build? If we choose not to build it at the point, we'd want to use a version that is close or identical to the larger one in the Tab Options Menu.

**Comment by merchako** [comment]
Indicators hidden and appear upon hover / focus / drag

![Image](https://github.com/paranext/paranext-core/assets/441789/9306b69b-42e2-4d14-b44a-486a10f70adf)

**Comment by Sebastian-ubs** [comment]
@merchako Do I understand correctly from this mockup, that the component used for scroll groups is the Shadcn `tab` component?

**Comment by merchako** [comment]
Incorrect; that was a custom component I designed. If we want something out-of-the-box for first iteration, we should pick something else. I expect the tab component might be confusing. It’s certainly too large for a pop up menu. @Sebastian-ubs

**Comment by Sebastian-ubs** [comment]
Which standard one (whith maybe adaptations) do you prefer?
I had added the following to the preview app on the examples page.
![grafik](https://github.com/user-attachments/assets/78971919-01da-427f-8024-a1d702e121fb)

like
```
<TabsList className="pr-h-7">
  <TabsTrigger className="pr-h-6 pr-w-0" value="a">
    A
  </TabsTrigger>
  <TabsTrigger className="pr-h-6 pr-w-0" value="b">
    B
  </TabsTrigger>
  <TabsTrigger className="pr-h-6 pr-w-0" value="c">
    C
  </TabsTrigger>
  <TabsTrigger className="pr-h-6 pr-w-0" value="d">
    D
  </TabsTrigger>
  <TabsTrigger className="pr-h-6 pr-w-0" value="-">
    -
  </TabsTrigger>
</TabsList>
```

**Comment by merchako** [comment]
> Which standard [Shadcn component] (with maybe adaptations) do you prefer?

The <[TabsList](https://ui.shadcn.com/docs/components/tabs)> you posted *looks* fine. I'm hesitant to endorse it since the semantics of the component contradict what it's being used for—it's not a tab.

The <[ToggleGroup](https://ui.shadcn.com/docs/components/toggle-group)> is more appealing semantically. There *is* a way to make them mutually exclusive. It's not apparent in the Shadcn docs, but in [Radix's ToggleGroup docs](https://www.radix-ui.com/primitives/docs/components/toggle-group#api-reference), you can demo how the `single` option makes the toggles mutually exclusive.

```
<ToggleGroup size={"sm"} type="single">
```

My vote is for _ToggleGroup_.

**Comment by tjcouch-sil** [comment]
[Shadcn demo of the Single option](https://ui.shadcn.com/docs/components/toggle-group#single) - seems pretty nice! Good find!
